Sources/DefaultsMacrosDeclarations/ObservableDefaultMacro.swift

```diff
@@ -0,0 +1,191 @@
+import SwiftCompilerPlugin
+import SwiftDiagnostics
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SwiftSyntaxMacros
+
+/**
+ Macro declaration for the ``ObservableDefault`` macro.
+*/
+public struct ObservableDefaultMacro {}
+
+/**
+Conforming to ``AccessorMacro`` allows us to add the property accessors (get/set) that integrate with ``Observable``.
+*/
+extension ObservableDefaultMacro: AccessorMacro {
+	public static func expansion(
+		of node: AttributeSyntax,
+		providingAccessorsOf declaration: some DeclSyntaxProtocol,
+		in context: some MacroExpansionContext
+	) throws(ObservableDefaultMacroError) -> [AccessorDeclSyntax] {
+		let property = try propertyPattern(of: declaration)
+		let expression = try keyExpression(of: node)
+		let associatedKey = associatedKeyToken(for: property)
+
+		// The get/set accessors follow the same pattern that @Observable uses to handle the mutations.
+		//
+		// The get accessor also sets up an observation to update the value when the UserDefaults
+		// changes from elsewhere. Doing so requires attaching it as an Objective-C associated
+		// object due to limitations with current macro capabilities and Swift concurrency.
+		return [
+			#"""
+			get {
+				if objc_getAssociatedObject(self, &Self.\#(associatedKey)) == nil {
+					let cancellable = Defaults.publisher(\#(expression))
+						.sink { [weak self] in
+							self?.\#(property) = $0.newValue
+						}
+					objc_setAssociatedObject(self, &Self.\#(associatedKey), cancellable, .OBJC_ASSOCIATION_RETAIN)
+				}
+				access(keyPath: \.\#(property))
+				return Defaults[\#(expression)]
+			}
+			"""#,
+			#"""
+			set {
+				withMutation(keyPath: \.\#(property)) {
+					Defaults[\#(expression)] = newValue
+				}
+			}
+			"""#
+		]
+	}
+}
+
+/**
+Conforming to ``PeerMacro`` we can add a new property of type Defaults.Observation that will update the original property whenever
+the UserDefaults value changes outside the class.
+*/
+extension ObservableDefaultMacro: PeerMacro {
+	public static func expansion(
+		of node: SwiftSyntax.AttributeSyntax,
+		providingPeersOf declaration: some SwiftSyntax.DeclSyntaxProtocol,
+		in context: some SwiftSyntaxMacros.MacroExpansionContext
+	) throws -> [SwiftSyntax.DeclSyntax] {
+		let property = try propertyPattern(of: declaration)
+		let associatedKey = associatedKeyToken(for: property)
+
+		return [
+			"private static var \(associatedKey): Void?"
```

Comment:
```suggestion
			"private nonisolated(unsafe) static var \(associatedKey): Void?"
```

This fixes the Swift 6 warning. It should be safe given that we only use it for its address.

Reply:
Oh my bad, for some reason I thought Swift 6 mode was enabled in the package by default.

I will update the macro, and thank you for pointing to the fix 😄